nav2_planner/src/planner_server.cpp

```diff
@@ -185,6 +185,8 @@ PlannerServer::computePathToPose()
 
     geometry_msgs::msg::PoseStamped start;
     if (!nav2_util::getCurrentPose(start, *tf_)) {
+      RCLCPP_INFO(get_logger(), "Start pose unavailable. Canceling planning action.");
```

Comment:
I don't think you should cancel a goal because 1 TF call was bad. Logging is good, but I think you should remove the terminate_goals

Reply:
I added the `terminate_goals` so that the planner will fail, otherwise the planner will return and the bt will continue to forever tick both the planner and the GoalChecker condition bt node which will forever keep checking for the tf transform and output an error. I don't think we necessarily want that behavior, but rather fail the navigation task when we don't have the pose? Maybe we could have a number of retries or a timeout on attempting to get the current pose?

Reply:
Ah got it, you're 100% right, I'm thinking like the planner action server is spinning and updating but its only updated when called.

I think having retries (1-2) would be beneficial in case somene's TF is running slower (ei only updates from encoders at <10hz or low compute so they're trying to skirt the minimum updates as possible) but not your problem in this PR